src/main/java/net/mcreator/workspace/misc/CreativeTabsOrder.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.workspace.misc;
+
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.parts.TabEntry;
+import net.mcreator.element.types.interfaces.ITabContainedElement;
+import net.mcreator.workspace.elements.ModElement;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CreativeTabsOrder extends ConcurrentHashMap<String, ArrayList<String>> {
+
+	public void addModElementToTabs(ModElement element) {
+		if (element.getGeneratableElement() instanceof ITabContainedElement tabElement) {
+			TabEntry tabEntry = tabElement.getCreativeTab();
+			if (tabEntry != null && !tabEntry.getUnmappedValue().equals("No creative tab entry") && this.containsKey(
+					tabEntry.getUnmappedValue()))
+				this.get(tabEntry.getUnmappedValue()).add(element.getName());
+		}
+	}
+
+	public void removeModElementFromTabs(ModElement element) {
+		if (element.getGeneratableElement() instanceof ITabContainedElement) {
+			for (ArrayList<String> tabContents : this.values())
+				tabContents.remove(element.getName());
+		}
+	}
+
+	public void updateModElementTabs(GeneratableElement element) {
+		if (element instanceof ITabContainedElement tabElement) {
+			TabEntry tabEntry = tabElement.getCreativeTab();
+			if (tabEntry == null || tabEntry.getUnmappedValue().equals("No creative tab entry"))
+				return;
+
+			// if order in new tab is overridden, add the element explicitly
+			String meName = element.getModElement().getName();
+			if (containsKey(tabEntry.getUnmappedValue()) && !this.get(tabEntry.getUnmappedValue()).contains(meName)) {

```

Comment:
It could be that order was overridden in previous tab, not current and in this case this could mean ME still needs to be removed from some other tab

Reply:
Tabs not overridden manually are not reflected in this map, so in that case the mod element would not be present there before this method was called.

Reply:
Yes, but this builds on premise current tab is overridden. What if user changed the tab and that other tab does not have order overridden, but previous one did. In this case, this if will be false and removal from old tab that had order overridden will not happen (similar thing to what I mentioned for empty tabs cases)

Reply:
That is being handled at this particular moment.

Reply:
(basically you need to check if any **but current tab** contains this ME and remove it)

Reply:
Please do some more testing after the change before marking for RFR again